### DISCUSSION TITLE
Deploying a webpage dp package and other server issues

### DISCUSSION THREAD

**Discussion by leonardoARoman** [eclipse-kura/kura#5080] [Q&A]
Hi, I recently installed Kura in a raspberry pi 3 and I noticed an issue I can't understand. I had a web app running using apache2 and NodeRed also running before Kura installation and now both servers can't be reached. I can see that both services are active and running but when I tried to reach the website from my workstation, the pages do not load/render. If I uninstall Kura however, everything resumes to work. Any thoughts? 

Second, how can I deploy the same web app within Kura. I can build a dp package using Angular build files and install them in the packages page in Kura. My deployment is set to copy all files in the "Dist" folder built by angular, into the dp bundle, the bundle then copies all those same files to /opt/kura/srv/www, and I set the security->Webconsole->(Web server Entry Point) field to redirect to something like "/login/ but it does not work. Is there a way to deploy a webpage as a bundle?

**Comment by salvatore-coppola** [comment]
>Hi, I recently installed Kura in a raspberry pi 3 and I noticed an issue I can't understand. I had a web app running using apache2 and NodeRed also running before Kura installation and now both servers can't be reached. I can see that both services are active and running but when I tried to reach the website from my workstation, the pages do not load/render. If I uninstall Kura however, everything resumes to work. Any thoughts?

Hi Leonardo, are you sure there are not conflicts between kura web ui console and your web apps tcp ports ?
Kura WebUI use by default the 443 and 4443 ports.

>Second, how can I deploy the same web app within Kura. I can build a dp package using Angular build files and install them in the packages page in Kura. My deployment is set to copy all files in the "Dist" folder built by angular, into the dp bundle, the bundle then copies all those same files to /opt/kura/srv/www, and I set the security->Webconsole->(Web server Entry Point) field to redirect to something like "/login/ but it does not work. Is there a way to deploy a webpage as a bundle?

You need to register a new servlet serving your resource.
Take a look at [Console](https://github.com/eclipse/kura/blob/f4994ee59d78d68f6a98cc04eb805ddc45039fe0/kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/api/Console.java) and [ClientExtensionBundle](https://github.com/eclipse/kura/blob/f4994ee59d78d68f6a98cc04eb805ddc45039fe0/kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/api/ClientExtensionBundle.java)

**Reply by leonardoARoman** [reply]
Hi, thanks for your reply. I was able to fix the first issue. It was actually a problem with the firewall. I just needed to open the ports used by the servers for apache and node red (80, 1880).

I will try your suggestion for the second question I asked. Thanks.

**Reply by mattdibi** [reply]
Does this work for you? If so we can close the discussion as answered.

**Comment by leonardoARoman** [comment]
I couldn’t deploy my costumed web UI. But I tried a different approach in
the raspberry PI. You can close this issue. Thanks.

V/R

Leonardo Roman
